include/nbl/asset/IRenderpass.h

```diff
@@ -13,102 +13,556 @@
+
+                    inline bool valid() const
+                    {
+                        auto disallowedFinalLayout = [](const auto& layout)->bool
+                        {
+                            switch (layout)
+                            {
+                                case IImage::EL_UNDEFINED: [[fallthrough]];
+                                case IImage::EL_PREINITIALIZED:
+                                    return true;
+                                    break;
+                                default:
+                                    break;
+                            }
+                            return false;
+                        };
+                        //
+                        if (disallowedFinalLayout(finalLayout))
+                            return false;
+                        //
+                        if (asset::isDepthOrStencilFormat(format) && !asset::isDepthOnlyFormat(format) && disallowedFinalLayout(finalLayout.actualStencilLayout()))
+                            return false;
+                        return true;
+                    }
+                    //inline bool used() const {return format!=EF_UNKNOWN;}
+                };
+                constexpr static inline SAttachmentDescription AttachmentsEnd = {};
+                const SAttachmentDescription* attachments = &AttachmentsEnd;
+            
+
+                struct SSubpassDescription
+                {
+                    constexpr static inline uint32_t AttachmentUnused = 0xffFFffFFu;
+                    template<typename layout_t>
+                    struct SAttachmentRef
+                    {
+                        public:
+                            // If you leave the `attachmentIndex` as default then it means its not being used
+                            uint32_t attachmentIndex = AttachmentUnused;
+                            layout_t layout = {};
+
+                            inline bool used() const {return attachmentIndex!=AttachmentUnused;}
+
+                            auto operator<=>(const SAttachmentRef<layout_t>&) const = default;
+                        
+                        protected:
+                            inline bool invalidLayout(const IImage::E_LAYOUT _layout)
+                            {
+                                switch (_layout)
+                                {
+                                    case IImage::EL_UNDEFINED: [[fallthrough]];
+                                    case IImage::EL_PREINITIALIZED: [[fallthrough]];
+                                    case IImage::EL_PRESENT_SRC:
+                                        return true;
+                                        break;
+                                    default:
+                                        break;
+                                }
+                                return false;
+                            }
+                            inline bool invalid() const
```

Comment:
It's weird to me sometimes you write a `bool invalid()` function for validation, and sometimes a `bool valid()`, even here in the same file, it gets hard to read and is inconsistent.
If there is any reasoning behind this, I'm curious

Reply:
I'd rather not have to remember to put a `!` in a bunch of if-statments, cause `!` is quite small and easy to overlook